File: python_tests/test_logging.py
"""Python-file for testing class Logging"""

from py_bindings import Logging, LogLevel

def test_logging():
    """Funktion zum Testen von LogLevel"""

    logger = Logging("loggingfile.txt", LogLevel.DEBUG, True)
    assert logger.get_level() == "DEBUG"
    logger.set_loglevel(LogLevel.ERROR)
    assert logger.get_level() == "ERROR"
    logger.set_loglevel(LogLevel.DEBUG)
    logger.enable_console_output(True)
    logger.start_log("System initialized.", LogLevel.INFO)
    logger.start_log("System deactivated.", LogLevel.INFO)
    logger.start_log("System restarted.", LogLevel.INFO)

File: cpp_tests/test_logging.cpp
#include "tests/test_logging.hpp"

void TestLogging() {
  Logging logger("logfile.txt", LogLevel::DEBUG);
  assert(std::filesystem::exists("logfile.txt"));
  assert(logger.GetLevel() == "DEBUG");
  logger.Log("System initialized.", LogLevel::INFO);
  logger.Log("Reading configuration...", LogLevel::DEBUG);
  logger.Log("File not found!", LogLevel::WARNING);
  logger.Log("Critical error!", LogLevel::ERROR);

  std::cout << "Alle Logging-Tests waren erfolgreich!" << std::endl;
}
